HIP/hip-1056.md

```diff
@@ -1707,6 +1693,77 @@ message RecordFileItem {
 
 ```
 
+#### TraceData
+
+A `TraceData` block item is added to contain all information that could be cosnidered trace or debugging in nature.
+The consensus node will export all necessary and important data for a transactions execution in the block streams,
+however every byte of information is stored fore er and needed in the proof of the network data. As such it's important
+to ensure that data is required but also laid out in a way that makes it easier for filtering by downstream clients.
+To this point the `TraceData` would initially contain smart contracts `EVMTraceData` that support traceability information (e.g.
+contract actions, read values etc). Future trace like data can be added to `TraceData` and Block Nodes, Mirror Nodes and
+other block stream parsign clients can decided to store or filter it out based on their needs.
+
+```protobuf
+message ContractSlotReads {
+     
+     message SlotRead {
+          oneof identifier {
+               /**
+               * The contract storage slot counter in this block 
+               * This is populated in place of the 256 bit word when the given slot is written to in state changes
+               */
+               int32 index = 1;
+
+               /**
+               * The key of this contratc storage slot, may be left-padded with zeros to form a 256-bit word.
+               * This is populated when the slot was not written and only read
+               */
+               bytes key = 2;
+          }
+
+          /**
+          * The storage value in this slot, may be left-padded with zeros to form a 256-bit word.
+          */
+          bytes read_value = 3;
+     }
+     
+     /**
+      * The contract assocaited with the storage slots this slot belongs to.
+      */
+     ContractID contractID = 1;
```

Comment:
snake case here and all other places

```suggestion
     ContractID contract_id = 1;
```

Reply:
Good note I was thinking of this too. Thanks

Reply:
Fixed